internal/api/v1/user/transactions/transactions_api.go

```diff
@@ -16,8 +17,58 @@ import (
 const route = "api/v1/transactions"
 
 type API struct {
-	url        *url.URL
-	httpClient *resty.Client
+	url               *url.URL
+	httpClient        *resty.Client
+	transactionSigner *transactionsigner.TransactionSigner
+}
+
+func (a *API) FinalizeTransaction(draft *response.DraftTransaction) (string, error) {
+	res, err := a.transactionSigner.GetSignedHex(draft)
+	if err != nil {
+		return "", fmt.Errorf("failed to finalize transaction: %w", err)
+	}
+
+	return res, nil
+}
+
+func (a *API) DraftToRecipients(ctx context.Context, r *commands.SendToRecipients) (*response.DraftTransaction, error) {
+	outputs := make([]*response.TransactionOutput, 0)
+
+	for _, recipient := range r.Recipients {
+		outputs = append(outputs, &response.TransactionOutput{
+			To:       recipient.To,
+			Satoshis: recipient.Satoshis,
+			OpReturn: recipient.OpReturn,
+		})
+	}
+
+	draftTransactionCmd := &commands.DraftTransaction{
+		Config: response.TransactionConfig{
+			Outputs: outputs,
+		},
+		Metadata: r.Metadata,
+	}
```

Comment:
The command could be passed directly as an argument in the `DraftTransaction` method call, which might streamline the code and improve clarity.